plugins/languages/cmake-tools.nix

```diff
@@ -0,0 +1,391 @@
+          The options the runner will get, possible values depend on the runner type. See `default_opts` for possible values.
+        '';
+
+        default_opts =
+          helpers.defaultNullOpts.mkAttributeSet ''
+            {
+              quickfix = {
+                show = "always"; # "always", "only_on_error"
+                position = "belowright"; # "bottom", "top"
+                size = 10;
+                encoding = "utf-8";
+                auto_close_when_success = true; # typically, you can use it with the "always" option; it will auto-close the quickfix buffer if the execution is successful.
+              };
+
+              toggleterm = {
+                direction = "float"; # 'vertical' | 'horizontal' | 'tab' | 'float'
+                close_on_exit = false; # whether close the terminal when exit
+                auto_scroll = true; # whether auto scroll to the bottom
+              };
+
+              overseer = {
+                new_task_opts = {
+                  strategy.__raw = '''
+                    {
+                      "terminal",
+                    },
+                  '''';
+                }; # options to pass into the `overseer.new_task` command
+                on_new_task = {
+                  __raw = '''
+                    function(task) end
+                  ''';
+                }; # a function that gets overseer.Task when it is created, before calling `task:start`
+              };
+
+              terminal = {
+                name = "Main Terminal";
+                prefix_name = "[CMakeTools]: "; # This must be included and must be unique, otherwise the terminals will not work. Do not use a simple spacebar " ", or any generic name
+                split_direction = "horizontal"; # "horizontal", "vertical"
+                split_size = 11;
+
+                # Window handling
+                single_terminal_per_instance = true; # Single viewport, multiple windows
+                single_terminal_per_tab = true; # Single viewport per tab
+                keep_terminal_static_location = true; # Static location of the viewport if available
+
+                # Running Tasks
+                start_insert = false; # If you want to enter terminal with :startinsert upon using :CMakeRun
+                focus = false; # Focus on terminal when cmake task is launched.
+              };
+            }
+          ''
+          ''
+            A list of default and possible values for runner.
+          '';
+      };
+
+    cmake_notifications =
+      helpers.mkCompositeOption ''
+        Notification settings.
+      ''
+      {
+        runner = helpers.mkCompositeOption "" {
+          enabled = helpers.defaultNullOpts.mkBool true "";
+        };
+
+        executor = helpers.mkCompositeOption "" {
+          enabled = helpers.defaultNullOpts.mkBool true "";
+        };
+
+        spinner = helpers.defaultNullOpts.mkListOf lib.types.str ''[ "⠋" "⠙" "⠹" "⠸" "⠼" "⠴" "⠦" "⠧" "⠇" "⠏" ]'' ''
```

Comment:
@traxys @GaetanLepage if `[ "⠋" "⠙" "⠹" "⠸" "⠼" "⠴" "⠦" "⠧" "⠇" "⠏" ]` is the [plugin's default](https://github.com/Civitasv/cmake-tools.nvim?tab=readme-ov-file#balloon-configuration), should _our_ default value just be `null`? How do we normally handle defaults like this?

Reply:
This is the right approach. The default provided to the `helpers.defaultNullOpts.*` helpers is the one of the plugin.
The actual default value of the nixvim option will always be `null`.